src/main/java/twilightforest/entity/EntityTFSlideBlock.java

```diff
@@ -158,20 +158,19 @@ public void tick() {
 				}
 
 				if (this.collided) {
-//					this.motionX *= 0.699999988079071D;
-//					this.motionZ *= 0.699999988079071D;
-//					this.motionY *= 0.699999988079071D;
-					this.getMotion().mul(0.699999988079071D, 0.699999988079071D, 0.699999988079071D);
+					// TODO: mul just returns a new Vec3d, it doesn't actually motify the speed
+					this.setMotion(this.getMotion().mul(0.699999988079071D, 0.699999988079071D, 0.699999988079071D));
 
 					this.remove();
 
 					if (this.world.canPlace(myState, pos, ISelectionContext.dummy())) {
 						world.setBlockState(pos, myState);
 					} else {
-						this.entityDropItem(new ItemStack(myState.getBlock(), 1, myState.getBlock().damageDropped(myState)), 0.0F);
+						// TODO: This and the below item might not be correctly reflecting the state
+						this.entityDropItem(new ItemStack(myState.getBlock()), 0.0F);
 					}
 				} else if (this.slideTime > 100 && (pos.getY() < 1 || pos.getY() > 256) || this.slideTime > 600) {
-					this.entityDropItem(new ItemStack(this.myState.getBlock(), 1, this.myState.getBlock().damageDropped(this.myState)), 0.0F);
+					this.entityDropItem(new ItemStack(this.myState.getBlock()), 0.0F);
```

Comment:
it should probably query the loot table.

Reply:
As it appears to be a "falling block" kind of block (except it's a sliding block), I wasn't sure if that value would be determinable inside of the loot table -- it's predicated in the constructor on line 48.

Reply:
Unless you meant the block loot table...